xformers/ops.py

```diff
@@ -86,17 +86,44 @@ class AttentionOpBase(torch.autograd.Function):
 
     FORWARD_OPERATOR: Any
     FORWARD_ERROR_ATOL: Mapping[torch.dtype, float] = {
-        torch.float: 2e-4,
-        torch.half: 2e-3,
+        torch.float: 3e-4,
+        torch.half: 4e-3,
         torch.bfloat16: 2e-3,
     }
+    FORWARD_ERROR_RTOL: Mapping[torch.dtype, float] = {
+        torch.float: 2e-5,
+        torch.half: 2e-5,
+        torch.bfloat16: 2e-5,
+    }
     SUPPORTED_DEVICES: Set[str]
     SUPPORTED_DTYPES: Set[torch.dtype]
     SUPPORTED_MAX_K: float
     SUPPORTED_ATTN_BIAS_TYPES: Set[Any] = {type(None)}
     SUPPORTS_DROPOUT: bool
     NAME: str
 
+    _TEST_BATCH_SIZES: List[int] = [1, 300]
+    _TEST_K: List[int] = [32, 128]
+
+    @classmethod
+    def generate_test_shapes_B_Mq_Mkv_K(cls):
```

Comment:
nit: I find it a bit weird that each operator has a method to generate test shapes. Not a blocker, but might be good to look into separating this afterwards, as we might also care about specific shapes for benchmarking